pilot/pkg/ambient/controller/waypoint_controller.go

```diff
@@ -147,19 +149,27 @@ func (rc *WaypointProxyController) Reconcile(name types.NamespacedName) error {
 		return nil
 	}
 
-	if gw.Spec.GatewayClassName != "istio-mesh" {
+	if gw.Spec.GatewayClassName != v1beta1.ObjectName(constants.WaypointGatewayClassName) {
 		log.Debugf("mismatched class %q", gw.Spec.GatewayClassName)
 		return nil
 	}
 
-	gatewaySA := gw.Annotations["istio.io/service-account"]
+	defaultName := istiogw.GetDefaultGatewayName(gw.Name, &gw.Spec)
+	proxyName := defaultName
+	if override, exists := gw.Annotations[istiogw.GatewayNameOverride]; exists {
+		proxyName = override
+	}
+
+	gatewaySA := gw.Annotations[constants.WaypointServiceAccount]
```

Comment:
@howardjohn how should `istio.io/service-account` and `gateway.istio.io/service-account` interact with each other.  Is `istio.io/service-account` just for populating the ForSA field?  And `gateway.istio.io/service-account` specifically for manually specifying the SA the waypoint proxy should use for itself?

Reply:
Yes, but, that is a bit confusing since the names are so similar. Wonder if we should make it `for-service-account` or something.... WDYT?

Reply:
works for me